### ISSUE TITLE
Please avoid the usage of if-else if tree

### ISSUE THREAD

**Issue by nwhean** [nwhean/42_core_minirt#5]
Instead of having a if-else tree like this:
https://github.com/nwhean/42_core_minirt/blob/4daabf14da385e9c6ce3da4adc5080178e93a0ba/src/error/error_handler.c#L86-L97

it will be better to have 2 arrays. One array containing the strings you want to match, another array containing the pointers to the corresponding functions. Then use a while loop to loop through them.

**Comment by HQLoke** [comment]
If I use 2 arrays, the function would be longer than expected and I would need an extra file because of the array initialisation. Do you still want me to proceed to change to the array method?

**Comment by nwhean** [comment]
Sorry for the slow reply. Can you copy pasta your code here and let me have a look?

**Comment by HQLoke** [comment]
This part only? Here you go.

static void	check_line(char *line, int line_num, t_list **error)
{
	char	**info;

	info = ft_split(line, ' ');
	if (ft_strcmp(info[0], "A") == 0)
		check_ambient(info, line_num, error);
	else if (ft_strcmp(info[0], "C") == 0)
		check_camera(info, line_num, error);
	else if (ft_strcmp(info[0], "L") == 0)
		check_light(info, line_num, error);
	else if (ft_strcmp(info[0], "sp") == 0)
		check_sphere(info, line_num, error);
	else if (ft_strcmp(info[0], "pl") == 0)
		check_plane(info, line_num, error);
	else if (ft_strcmp(info[0], "cy") == 0)
		check_cylinder(info, line_num, error);
	else
		ft_lstadd_back(error, ft_lstnew("Invalid type identifier", line_num));
	ft_memdel((void **)info);
}

**Comment by nwhean** [comment]
I mean your amended code that uses array.

**Comment by HQLoke** [comment]
```
int		i;
	char	**info;
	char	**spec;
	void	(*f[6])(char **, int, t_list**);

	info = ft_split(line, ' ');
	spec = ft_split("A C L sp pl cy", ' ');
	f[0] = check_ambient;
	f[1] = check_camera;
	f[2] = check_light;
	f[3] = check_sphere;
	f[4] = check_plane;
	f[5] = check_cylinder;
	i = 0;
	while (spec[i] != NULL)
	{
		if (ft_strcmp(info[0], spec[i]) == 0)
		{
			f[i](info, line_num, error);
			break ;
		}
		i += 1;	
	}
	if (i == 6)
		ft_lstadd_back(error, ft_lstnew("Invalid type identifier", line_num));
	ft_memdel((void **)info);
	ft_memdel((void **)spec);
```

**Comment by nwhean** [comment]
Resolved in commit 91a137d15a8105e9b48463a9e376001a2c1773f5.
